File: 2º/PDOO/ProyectoRUBY/lib/c_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "casilla.rb"
require_relative "jugador.rb"

module Civitas
  class C_sorpresa<Casilla
    def initialize(mazo, nombre)
      super(nombre)
      @mazo=mazo
    end
    def  to_s
      devolver=super()
      devolver+="\n}\n"
      return devolver;
    end  
    def recibeJugador(iactual, todos)
      if(jugadorCorrecto(iactual,todos))then
        sorpresa=@mazo.siguiente
        super.informe(iactual,todos)
        sorpresa.aplicarAJugador(iactual, todos)
      end
    end

  end
end

File: 2º/PDOO/ProyectoJAVA/src/civitas/C_sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class C_sorpresa extends Casilla{
    
    private Sorpresa sorpresa;
    private MazoSorpresas mazo;
    
    C_sorpresa(MazoSorpresas mazo, String nombre) {         // sorpresa
        super(nombre);
        this.mazo = mazo;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            Sorpresa sorpresa = mazo.siguiente();
            informe(actual, todos);
            sorpresa.aplicarAJugador(actual, todos);
        }
    }
    
    @Override
    public String toString() {
        String e = "\n-> Casilla ("+ super.getNombre() + ")";
        return e;
    }
}
